src/main/java/com/dabsquared/gitlabjenkins/util/CommitStatusUpdater.java

```diff
@@ -64,7 +72,15 @@ private static void printf(TaskListener listener, String message, Object... args
     }
 
     private static String getBuildRevision(Run<?, ?> build) {
-        return build.getAction(BuildData.class).getLastBuiltRevision().getSha1String();
+        BuildData action = build.getAction(BuildData.class);
+        try {
+            Field lastBuildField = action.getClass().getDeclaredField("lastBuild");
+            lastBuildField.setAccessible(true);
+            Build lastBuild = (Build) lastBuildField.get(action);
+            return lastBuild.getMarked().getSha1String();
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            return null;
+        }
     }
```

Comment:
lastBuild is a public field of BuildData, and up to now there was no null checking for the result of `getLastBuiltRevision`, so it seems to be always setl.
https://github.com/jenkinsci/git-plugin/blob/7ea2ad06da30aa6778b33fdd4c1f94389c20fbee/src/main/java/hudson/plugins/git/util/BuildData.java#L41